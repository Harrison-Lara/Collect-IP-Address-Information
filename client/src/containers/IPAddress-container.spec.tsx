import React from 'react';
import { shallow } from 'enzyme';
import { IPAddress, validate } from './IPAddress-container';

describe('IPAddress', () => {
  it('Render the IPAddress container', () => {
    const component = shallow(<IPAddress />);

    expect(component).toHaveLength(1);
    expect(component).toMatchSnapshot();
  });

  it('validate with an empty value', () => {
    const emptyAddress = '';

    expect(validate(emptyAddress)).toEqual({ "address": "Invalid IP (Ex. 8.8.8.8)" })
    expect(validate(132)).toEqual({ "address": "Invalid IP (Ex. 8.8.8.8)" })
  });
});