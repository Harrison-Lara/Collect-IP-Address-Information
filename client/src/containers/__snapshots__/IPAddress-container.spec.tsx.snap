// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IPAddress Render the IPAddress container 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <WithStyles(ForwardRef(Grid))
      container={true}
      spacing={8}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={4}
      />
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={4}
      >
        <WithStyles(ForwardRef(Typography))
          align="center"
          className="makeStyles-title-1"
          variant="h3"
        >
          IP Information
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(FormGroup))>
          <TextField
            error={false}
            helperText="Ex. 8.8.8.8"
            id="address"
            label="IP Address"
            onBlur={[Function]}
            onChange={[Function]}
            value=""
          />
          <Button
            className="makeStyles-buttonMargin-2"
            color="primary"
            onClick={[Function]}
            variant="contained"
          >
            Search
          </Button>
        </WithStyles(ForwardRef(FormGroup))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </form>
  <WithStyles(ForwardRef(Grid))
    className="makeStyles-panelWidth-3"
    item={true}
    xs={10}
  >
    <WithStyles(ForwardRef(ExpansionPanel))
      defaultExpanded={true}
    >
      <WithStyles(ForwardRef(ExpansionPanelSummary))
        expandIcon={<UNDEFINED />}
        id="detail-panel"
      >
        <WithStyles(ForwardRef(Typography))
          variant="h6"
        >
          Retrieved Information
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(ExpansionPanelSummary))>
      <WithStyles(ForwardRef(ExpansionPanelDetails))>
        <WithStyles(ForwardRef(Typography))
          variant="body1"
        >
          Awaiting Input...
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(ExpansionPanelDetails))>
    </WithStyles(ForwardRef(ExpansionPanel))>
  </WithStyles(ForwardRef(Grid))>
</Fragment>
`;
